modules/greenbidsBidAdapter.js

```diff
@@ -0,0 +1,272 @@
+}
+
+function getTimeToFirstByte(win) {
+  const performance = win.performance || win.webkitPerformance || win.msPerformance || win.mozPerformance;
+
+  const ttfbWithTimingV2 = performance &&
+    typeof performance.getEntriesByType === 'function' &&
+    Object.prototype.toString.call(performance.getEntriesByType) === '[object Function]' &&
+    performance.getEntriesByType('navigation')[0] &&
+    performance.getEntriesByType('navigation')[0].responseStart &&
+    performance.getEntriesByType('navigation')[0].requestStart &&
+    performance.getEntriesByType('navigation')[0].responseStart > 0 &&
+    performance.getEntriesByType('navigation')[0].requestStart > 0 &&
+    Math.round(
+      performance.getEntriesByType('navigation')[0].responseStart - performance.getEntriesByType('navigation')[0].requestStart
+    );
+
+  if (ttfbWithTimingV2) {
+    return ttfbWithTimingV2.toString();
+  }
+
+  const ttfbWithTimingV1 = performance &&
+    performance.timing.responseStart &&
+    performance.timing.requestStart &&
+    performance.timing.responseStart > 0 &&
+    performance.timing.requestStart > 0 &&
+    performance.timing.responseStart - performance.timing.requestStart;
+
+  return ttfbWithTimingV1 ? ttfbWithTimingV1.toString() : '';
+}
+
+function cleanBidsInfo(bids) {
+  const reqObj = {};
+  let placementId = getValue(bids.params, 'placementId');
+  const gpid = deepAccess(bids, 'ortb2Imp.ext.gpid');
+  reqObj.sizes = getSizes(bids);
+  reqObj.bidId = getBidIdParameter('bidId', bids);
+  reqObj.bidderRequestId = getBidIdParameter('bidderRequestId', bids);
+  reqObj.placementId = parseInt(placementId, 10);
+  reqObj.adUnitCode = getBidIdParameter('adUnitCode', bids);
+  reqObj.transactionId = bids.ortb2Imp?.ext?.tid || '';
+  if (gpid) { reqObj.gpid = gpid; }
+  return reqObj;
+}
+
+function getSizes(bid) {
+  return parseSizesInput(bid.sizes);
+}
+
+// Privacy handling
+
+export function hydratePayloadWithGPPData(payload, gppData) {
+  if (gppData) {
+    let isValidConsentString = typeof gppData.gppString === 'string';
+    let validateApplicableSections =
+      Array.isArray(gppData.applicableSections) &&
+      gppData.applicableSections.every((section) => typeof (section) === 'number')
+    payload.gpp = {
+      consentString: isValidConsentString ? gppData.gppString : '',
+      applicableSectionIds: validateApplicableSections ? gppData.applicableSections : [],
+    };
+  }
+  return payload;
+}
+
+export function hydratePayloadWithGDPRData(payload, gdprData) {
+  if (gdprData) {
+    let isCmp = typeof gdprData.gdprApplies === 'boolean';
+    let isConsentString = typeof gdprData.consentString === 'string';
+    let status = isCmp
+      ? findGdprStatus(gdprData.gdprApplies, gdprData.vendorData)
+      : gdprStatus.CMP_NOT_FOUND_OR_ERROR;
+    payload.gdpr_iab = {
+      consent: isConsentString ? gdprData.consentString : '',
+      status: status,
+      apiVersion: gdprData.apiVersion
+    };
+  }
```

Comment:
suggestion: to avoid one wide-spread nesting level, you could revert the condition:
```
  if (!gdprData) { return ; }
  let isCmp = ...
  ...
```